lib/resourceful_endpoint.js

```diff
@@ -212,7 +212,67 @@ class ResourcefulEndpoint {
    * @returns {Promise} - A {@link ResourceCollection}
    */
   browse(criteria, options) {
-    return this.transport.get(this.endPointUrl(null, criteria), options).then(json => new ResourceCollection(json, criteria, this));
+    return this.transport
+      .get(this.endPointUrl(null, this.getRelatedThroughFields(criteria)), options)
+      .then(json => new ResourceCollection(json, criteria, this));
+  }
+
+  /**
+   * Private
+   * Look through the relationships of a criteria, and include fields from through
+   * relationships - this makes it so that the end user shouldn't know or care that
+   * through relationships exist.
+   *
+   * @param {(string|Query)} criteria - A query string to append to the request
+   *
+   * @returns {(Query)} criteria - A qquery with additional includes if needed
+
+   */
+  getRelatedThroughFields(criteria) {
+    if (!criteria || !criteria.query) return criteria;
+    const query = criteria.query || '';
+    const params = query.split('&');
+    const includes = params.find(param => param.startsWith('include=') || '')
+      .replace('include=', '')
+      .split(',');
+
+    let fields = params.find(param => param.startsWith('fields=')) || '';
```

Comment:
Does this section work as expected?  When splitting, the first entry will have `fields=` at the start, so would result in a double `fields=` later on?

Reply:
I'll check this stuff if I get the nod for my proposed approach